honeybee_radiance/state.py

```diff
@@ -0,0 +1,744 @@
+
+    def move(self, moving_vec):
+        """Move all shades assigned to this state along a vector.
+
+        Args:
+            moving_vec: A ladybug_geometry Vector3D with the direction and distance
+                to move the shades.
+        """
+        for shd in self._shades:
+            shd.move(moving_vec)
+
+    def rotate(self, axis, angle, origin):
+        """Rotate all shades assigned to this state.
+
+        Args:
+            axis: A ladybug_geometry Vector3D axis representing the axis of rotation.
+            angle: An angle for rotation in degrees.
+            origin: A ladybug_geometry Point3D for the origin around which the
+                object will be rotated.
+        """
+        for shd in self._shades:
+            shd.rotate(axis, angle, origin)
+
+    def rotate_xy(self, angle, origin):
+        """Rotate all shades counterclockwise in the world XY plane.
+
+        Args:
+            angle: An angle in degrees.
+            origin: A ladybug_geometry Point3D for the origin around which the
+                object will be rotated.
+        """
+        for shd in self._shades:
+            shd.rotate_xy(angle, origin)
+
+    def reflect(self, plane):
+        """Reflect all shades assigned to this state across a plane.
+
+        Args:
+            plane: A ladybug_geometry Plane across which the object will
+                be reflected.
+        """
+        for shd in self._shades:
+            shd.reflect(plane)
+
+    def scale(self, factor, origin=None):
+        """Scale all shades assigned to this state by a factor.
+
+        Args:
+            factor: A number representing how much the object should be scaled.
+            origin: A ladybug_geometry Point3D representing the origin from which
+                to scale. If None, it will be scaled from the World origin (0, 0, 0).
+        """
+        for shd in self._shades:
+            shd.scale(factor, origin)
+
+    def to_radiance(self, direct=False, minimal=False):
+        """Generate an RAD string representation of this state.
+
+        Note that the resulting string lacks modifiers but includes both the
+        parent geometry and the geometry of any shades.
+
+        Args:
+            direct: Boolean to note whether to write the "direct" version of the
+                state, which will have the modifier_direct applied to the parent
+                of the state and use the modifier_blk assigned to each of the
+                shades. (Default: False)
+            minimal: Boolean to note whether the radiance string should be written
+                in a minimal format (with spaces instead of line breaks). Default: False.
+        """
+        assert self.has_parent, 'State must have a parent to use to_radiance.'
+        modifier = self.modifier_direct if direct else self.modifier
+        base_poly = Polygon(self.parent.identifier, self.parent.vertices, modifier)
+        rad_strs = [base_poly.to_radiance(minimal, False, False)]
```

Comment:
This type of comment in the .rad string is a good idea but I don't think this is the right place for it since this method is likely going to return a single polygon for the vast majority of cases. Rather, the correct place for this type of comment is in the `DynamicGroup.to_radiance` method, which is in charge of writing out each .rad file of the group's state.

Reply:
I added a bunch of comments in the `DynamicGroup.to_radiance` methods so that each .rad file gets a comment at the top.